File: poo/java/Interfaces/Carbono/Car.java
/*
 *  Programa escrito com propósito didático apenas.
 *  @author Nathan Cruz <nathan.cruz@unifesp.br>
 *  @created 04/04/14
 *  R.A.: 86492
 */

public class Car implements CarbonFootprint{

	/* Atributos */
	private String modelo;
	private String combustivel;
	private float cilindrada;
	private int passageiros;

	/* Construtores */
	public Car(String modelo, String combustivel,
				float cilindrada, int passageiros){
		this.modelo = modelo;
		this.combustivel = combustivel;
		this.cilindrada = cilindrada;
		this.passageiros = passageiros;
	}
	public Car(){
		this.modelo = "Corsa";
		this.combustivel = "Alcool";
		this.cilindrada = 1000;
		this.passageiros = 4;
	}

	/* Implementacao da interface */
	public double getCarbonFootprint(){
		return this.cilindrada*0.7 + this.passageiros*0.1;
	}

	public String toString(){
		return "\nTipo: Car\nQuantidade de passageiros: " + this.passageiros + "\nCombustivel: " +
				this.combustivel + "\nModelo: " + this.modelo + "\nCilindradas: "+
				this.cilindrada + "\nPegada de Carbono: " + this.getCarbonFootprint();
	}

}

File: poo/python/Interfaces/Carbono/Car.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 12/04/14
"""
from __future__ import unicode_literals
from __future__ import print_function

class Car():

    # Atributos #
    __modelo = None
    __combustivel = None
    __cilindrada = None
    __passageiros = None

    # Construtor #
    def __init__(self, modelo='Corsa', combustivel='Alcool',
                 cilindrada=1000, passageiros=4):
        self.__modelo = modelo
        self.__combustivel = combustivel
        self.__cilindrada = cilindrada
        self.__passageiros = passageiros

    def getCarbonFootprint(self):
        return (self.__cilindrada * 0.7 +
                self.__passageiros * 0.1)

    def __str__(self):
        return '''
                Tipo: Car
                Quantidade de passageiros: {0}
                Combustivel: {1}
                Modelo: {2}
                Cilindradas: {3}
                Pegada de carbono: {4}'''.format(self.__passageiros,
                                                 self.__combustivel,
                                                 self.__modelo,
                                                 self.__cilindrada,
                                                 self.getCarbonFootprint())

    def __repr__(self):
        return self.__str__()
